fhir-persistence/src/test/java/com/ibm/fhir/persistence/test/common/AbstractCanonicalTest.java

```diff
@@ -0,0 +1,769 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.persistence.test.common;
+
+import static com.ibm.fhir.model.type.String.string;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.ibm.fhir.model.format.Format;
+import com.ibm.fhir.model.resource.CarePlan;
+import com.ibm.fhir.model.resource.Library;
+import com.ibm.fhir.model.resource.Measure;
+import com.ibm.fhir.model.resource.Resource;
+import com.ibm.fhir.model.test.TestUtil;
+import com.ibm.fhir.model.type.Canonical;
+import com.ibm.fhir.model.type.Reference;
+import com.ibm.fhir.model.type.Uri;
+import com.ibm.fhir.model.type.code.ResourceType;
+
+/**
+ *  This class tests the persistence layer support for the FHIR canonical search parameters.
+ */
+public abstract class AbstractCanonicalTest extends AbstractPersistenceTest {
+    private static CarePlan savedCarePlan1;
+    private static CarePlan savedCarePlan2;
+    private static CarePlan savedCarePlan3;
+    private static Measure savedMeasure1;
+    private static Measure savedMeasure2;
+    private static Measure savedMeasure3;
+    private static Measure savedMeasure4;
+    private static Library savedLibrary1;
+    private static Library savedLibrary2;
+    private static Library savedLibrary3;
+    private static String uuid = UUID.randomUUID().toString();
+
+    /**
+     * Loads up and saves a bunch of resources with various canonical references to one another
+     */
+    @BeforeClass
+    public void createResources() throws Exception {
+        CarePlan carePlan = TestUtil.getMinimalResource(ResourceType.CARE_PLAN, Format.JSON);
+        Measure measure = TestUtil.getMinimalResource(ResourceType.MEASURE, Format.JSON);
+        Library library = TestUtil.getMinimalResource(ResourceType.LIBRARY, Format.JSON);
```

Comment:
I broke these with https://github.com/IBM/FHIR/pull/2701.  Now it will be:
```suggestion
        CarePlan carePlan = TestUtil.getMinimalResource(CarePlan.class);
        Measure measure = TestUtil.getMinimalResource(Measure.class);
        Library library = TestUtil.getMinimalResource(Library.class);
```

Reply:
done